lib/src/Timer/Timer.tsx

```diff
@@ -6,80 +6,67 @@ export type TimerProps = JSX.IntrinsicElements["div"] & {
-    private innerInterval: any;
-    constructor(props: TimerProps) {
-        super(props);
+export const Timer: React.FC<TimerProps> = ({ duration, callback, timerPrefix, timerSuffix, ...props }: TimerProps) => {
+    const [innerInterval, setInnerInterval] = React.useState<NodeJS.Timeout>();
+    const [timer, setTimer] = React.useState<string>("00:00");
 
-        this.state = {
-            timer: "00:00",
-        };
-    }
-
-    startInterval(timeout: number): void {
-        this.setState({ timer: this.convertMStoTime(timeout) }, () => {
-            this.clearInterval();
-            this.innerInterval = setInterval(() => {
+    const startInterval = (timeout: number): void => {
+        setTimer(convertMStoTime(timeout));
+        clearInterval();
+        setInnerInterval(
+            setInterval(() => {
                 if (timeout > 0) {
                     timeout = timeout - 1000;
-                    this.setState({ timer: this.convertMStoTime(timeout) }, () => {
-                        if (timeout === 0) {
-                            this.props.callback();
-                            this.clearInterval();
-                        }
-                    });
+                    setTimer(convertMStoTime(timeout));
+                    if (timeout === 0) {
+                        callback?.();
+                        clearInterval();
+                    }
                 }
-            }, 1000);
-        });
-    }
-
-    clearInterval(): void {
-        if (this.innerInterval) {
-            clearTimeout(this.innerInterval);
-            this.innerInterval = null;
-        }
-    }
-
-    convertMStoTime(value: number): string {
-        const date: Date = new Date(value);
-        return (
-            (date.getUTCHours() > 0 ? date.getUTCHours() + ":" : "") +
-            (date.getUTCMinutes() < 10 ? "0" + date.getUTCMinutes() : date.getUTCMinutes()) +
-            ":" +
-            (date.getUTCSeconds() < 10 ? "0" + date.getUTCSeconds() : date.getUTCSeconds())
+            }, 1000)
         );
-    }
+    };
 
-    componentDidMount() {
-        if (this.props.duration !== null && this.props.duration !== undefined) {
-            this.startInterval(this.props.duration);
+    const clearInnerInterval = (): void => {
+        if (innerInterval) {
+            clearInterval(innerInterval);
+            setInnerInterval(null);
         }
-    }
+    };
 
-    componentWillUnmount() {
-        this.clearInterval();
-    }
+    const convertToTwoDigits = (rawNumber: number): string => `${rawNumber < 10 ? "0" : ""}${rawNumber}`;
 
-    componentDidUpdate(prevProps: TimerProps): void {
-        if (prevProps.duration !== this.props.duration) {
-            if (this.props.duration !== null && this.props.duration !== undefined) {
-                this.startInterval(this.props.duration);
-            }
+    const convertMStoTime = (milliseconds: number): string => {
```

Comment:
maybe we can rename this to `formatTime`? generally in date-time related libraries, they reserve `convertXX` for conversion between time units and `formatXyz` for display related transformation

Reply:
or maybe even `formatMilitaryTime`

Reply:
this algorithm seemed to work as well
```
return new Date(milliseconds).toISOString().slice(11, 19);
```

Reply:
rename function